### PR TITLE
Add bash completion for `docker stack`

### PR DESCRIPTION
**Pull request by albers**
Ref: #28677
Please schedule for 1.13.0
Ping @sdurrheimer for zsh completion

Note:  `docker stack config` is not available yet, so I did not add completion:
```bash
$ docker help stack

Usage:  docker stack COMMAND

Manage Docker stacks

Options:
      --help   Print usage

Commands:
  deploy      Deploy a new stack or update an existing stack
  ls          List stacks
  ps          List the tasks in the stack
  rm          Remove the stack
  services    List the services in the stack

Run 'docker stack COMMAND --help' for more information on a command.
```

### CODE REVIEW

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -3120,6 +3136,166 @@ _docker_search() {
...
+	esac
+}
+
+_docker_stack_deploy() {
+	case "$prev" in
+		--bundle-file)
+			_filedir dab
+			return
+			;;
+		--compose-file)
```
Comment (thaJeztah) [inline contrib/completion/bash/docker]: heads up: we'll be adding a shorthand `-c` flag for `--compose-file` (see https://github.com/docker/docker/pull/28815)

**Comment by thaJeztah** [comment]
ping @vdemeester @dnephin please review :)

**Review by dnephin — APPROVED** [review]

**Comment by sdurrheimer** [comment]
What's the state of `config` and `services` subcommands ? Both are marked as `experimental` but only `services` is displayed in the `--help` output as @albers pointed it above.

Should we add them in the completion ? Both ?, Only `services` ?

**Comment by albers** [comment]
Added `-c`, see #28815

**Comment by dnephin** [comment]
`config` as a subcommand has been removed

**Comment by sdurrheimer** [comment]
@dnephin Thx, in this case, the commandline reference documentation `docs/reference/commandline/stack_config.md` should be removed @thaJeztah.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**